From d3daf73a2d58c9f0ba32d7a42254a0cc0f61475e Mon Sep 17 00:00:00 2001
From: Flopster101 <nahuelgomez329@gmail.com>
Date: Mon, 23 Feb 2026 03:47:47 -0300
Subject: [PATCH] android: replace lockf with fcntl lock on Android

---
 src/frontends/common/songdb.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/frontends/common/songdb.c b/src/frontends/common/songdb.c
index 7eefe86..e953b34 100644
--- a/src/frontends/common/songdb.c
+++ b/src/frontends/common/songdb.c
@@ -216,7 +216,17 @@ static int uade_open_and_lock(const char *filename, int create)
 		}
 	}
 #ifndef UADE_HAVE_CYGWIN
+#ifdef __ANDROID__
+	struct flock lock_spec;
+	memset(&lock_spec, 0, sizeof(lock_spec));
+	lock_spec.l_type = F_WRLCK;
+	lock_spec.l_whence = SEEK_SET;
+	lock_spec.l_start = 0;
+	lock_spec.l_len = 0; /* lock whole file */
+	ret = fcntl(fd, F_SETLKW, &lock_spec);
+#else
 	ret = lockf(fd, F_LOCK, 0);
+#endif
 	if (ret) {
 		fprintf(stderr, "uade: Unable to lock song.conf: %s (%s)\n",
 			filename, strerror(errno));
-- 
2.53.0

